gnucash/report/gnc-report.cpp

```diff
@@ -262,19 +263,20 @@ gnc_report_id_string_to_report_id (const char *id_string)
 {
     g_return_val_if_fail (id_string, -1);
 
-    char *end_ptr;
-    guint rpt_id = std::strtoul (id_string, &end_ptr, 10);
-    if (end_ptr == id_string) return -1;
-    if (*end_ptr == '\0') return rpt_id;
-    if (*end_ptr != '|') return -1;
+    const char *end_ptr = id_string + std::strlen (id_string);
+    gint rpt_id;
```

Comment:
You don't seem to expect negative report id's. So why wouldn't you then make `rpt_id` a `guint` (or `uint32_t` for that matter) ?
I may read the documentation on `std::from_chars` wrongly (as I did with `std::strtoul`), but I **think** it has overloads for all common integer types. It seems to me `uint23_t` is a valid integer type, no ?

Reply:
Ok the motivation is to minimise the need for casting and tests. If we make `rpt_id` a `uint` and the function returns an `int`, then we need to range check `uint` before returning `int`. I'd rather make the range check takes place within `from_chars` which will then error out appropriately.

Reply:
I ignored the purpose of the negative return value indeed.

On a related note I recently read about [std::expected](https://en.cppreference.com/w/cpp/utility/expected.html) as the perfect fit for this use case. Unfortunately it's C++23 only, so unfortunately we can't use that one yet...